src/order/filter/n_ary.lean

```diff
@@ -49,6 +50,29 @@ def map₂ (m : α → β → γ) (f : filter α) (g : filter β) : filter γ :=
 lemma image2_mem_map₂ (hs : s ∈ f) (ht : t ∈ g) : image2 m s t ∈ map₂ m f g :=
 ⟨_, _, hs, ht, subset.rfl⟩
 
+lemma map_prod_eq_map₂ {f : filter α} {g : filter β} (m : α → β → γ) :
+  filter.map (λ p : α × β, m p.1 p.2) (f ×ᶠ g) = map₂ m f g :=
```

Comment:
It might also be worth having a variant:
```lean
  filter.map m (f ×ᶠ g) = map₂ (λ a b, m (a, b)) f g :=
```

Reply:
Is it? All the binary operations I have seen have the form m : α → β → γ.

Reply:
The point is that my version is strictly more general when rewriting in the forwards direction - it matches `filter.map (uncurry h) (f ×ᶠ g)` syntactically, while your version does not.

Reply:
I have no knowledge about how lean does rewriting, so I'll take your word for it. What should I call your variant? `map_prod_eq_map₂'`?

Reply:
That sounds good to me